tests/browser/test/procedure_test.js

```diff
@@ -92,19 +88,16 @@ suite('Testing Connecting Blocks', function (done) {
       'Functions',
       4
     );
-    await new Promise((resolve) => setTimeout(resolve, 2000)); // 2 sec
     await doSomething2Flyout.dragAndDrop({x: 130, y: 20});
-    await new Promise((resolve) => setTimeout(resolve, 2000)); // 2 sec
     const doSomething2Caller = await getSelectedBlockElement(browser);
 
     // Connect doSomething2 caller with print.
     await connect(browser, doSomething2Caller, 'OUTPUT', print, 'TEXT');
 
     // Click run button and verify the number is 123
-    await new Promise((resolve) => setTimeout(resolve, 2000)); // 2 sec
     const runButton = await browser.$('#runButton');
     runButton.click();
-    await new Promise((resolve) => setTimeout(resolve, 2000)); // 2 sec
+    await sleep(2000); // 2s
```

Comment:
1. Would a shorter sleep work reliably enough?
2. Is there another way to wait for execution to finish, that would detect it immediately?  (Could be topic for subsequent PR.)